src/app/hooks/useWindowScroll.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react'
+
+export const useWindowScroll = (): number => {
+  const [windowScrollY, setWindowScrollY] = useState(0)
+
+  useEffect(() => {
+    const windowScrollListener = () => {
+      setWindowScrollY(window.scrollY)
+    }
+
+    window.addEventListener('scroll', windowScrollListener)
```

Comment:
Scroll should almost always be a passive listener

Reply:
Right, it's required, because offset calculation depends on the scroll. So that scroll actually triggers it.

Reply:
I don't understand 
```suggestion
    window.addEventListener('scroll', windowScrollListener, { passive: true })
```

Reply:
Ah, I thought you meant it should not trigger CD. Nevermind. Let's discuss on sync-up tomorrow.